### DISCUSSION TITLE
NEW iOS 15 Ad hoc build will not install

### DISCUSSION THREAD

**Discussion by Tron307020** [airsdk/Adobe-Runtime-Support#1297] [Q&A]
I am experiencing a problem with Ad hoc distribution on iOS 15. Upon an attempted install (ad-hoc), I am receiving the error: "[APP] Needs To Be Updated" (bold title) and "The developer of this app needs to update it to work with this version of iPadOS." Previous iOS versions work fine. I am using the latest SDK 33.1.1.633.

I had thought that updating the adt.jar file would work as it did for others here, but (1) none of the links to download the patch work anymore and (2) it was noted that the updated adt.jar file would be in release 620 and so also 633.

Any thoughts?

**Comment by ajwfrost** [comment]
If you're using the .633 build of the SDK then it should have everything it needs for the code-signing requirements of iOS15 plus the SDK/Xcode version information.... with the exception that there are some problems with the code signatures of dynamic frameworks, but the error outputs you mention above are not what we'd expect if that was the cause.

One thing I am wondering is how new your certificates are as the Apple intermediate certificates changed a while ago - are you able to check the issuer details of your certificates to ensure they're the latest Apple (G3) versions?

If you're able to unzip your application (IPA file) on a mac and run `codesign -d -vvvv Payload/appname.app` on this from a terminal window, we can perhaps look at the output and see if anything else is obvious here.?

thanks

**Reply by Tron307020** [reply]
Actually, this was entirely user (my) error. The latest SDK version (633) I downloaded worked perfectly fine when compiling for and installing on iOS 15. The server was delivering an older, cached, copy of the application to the new devices and that was built with a 4xx version of the SDK.

Sorry for the confusion and thanks for the reply. This thread should be closed and deleted, if possible.
